synapse/replication/tcp/handler.py

```diff
@@ -87,6 +89,12 @@ def __init__(self, hs):
                 self._streams_to_replicate.append(stream)
                 continue
 
+            if (
+                isinstance(stream, (EventsStream, BackfillStream))
+                and hs.config.worker.writers.events == hs.get_instance_name()
+            ):
+                self._streams_to_replicate.append(stream)
```

Comment:
A comment for why we do this would be helpful.

Reply:
Done, let me know if it makes sense